src/test/java/org/opensearch/sdk/TestSDKTransportService.java

```diff
@@ -101,4 +107,43 @@ public void testRegisterTransportAction() {
         // Internal action should be filtered out
         assertFalse(registerTransportActionsRequestCaptor.getValue().getTransportActions().contains(RemoteExtensionAction.class.getName()));
     }
+
+    @Test
+    public void testRemoteExtensionActionRequest() {
+        ArgumentCaptor<TransportActionRequestFromExtension> transportActionRequestFromExtensionCaptor = ArgumentCaptor.forClass(
+            TransportActionRequestFromExtension.class
+        );
+        String expectedAction = "com.example.action";
+        String expectedRequest = "com.example.request";
+        byte[] expectedRequestBytes = "test".getBytes(StandardCharsets.UTF_8);
+        RemoteExtensionActionRequest request = new RemoteExtensionActionRequest(expectedAction, expectedRequest, expectedRequestBytes);
+        sdkTransportService.sendRemoteExtensionActionRequest(request);
+        verify(transportService, times(1)).sendRequest(
+            any(),
+            eq(ExtensionsManager.TRANSPORT_ACTION_REQUEST_FROM_EXTENSION),
+            transportActionRequestFromExtensionCaptor.capture(),
+            any(ExtensionActionResponseHandler.class)
+        );
+        assertEquals(TEST_UNIQUE_ID, transportActionRequestFromExtensionCaptor.getValue().getUniqueId());
+        assertEquals(expectedAction, transportActionRequestFromExtensionCaptor.getValue().getAction());
+        String expectedString = expectedRequest + (char) RemoteExtensionActionRequest.UNIT_SEPARATOR + "test";
+        assertEquals(
+            expectedString,
+            new String(transportActionRequestFromExtensionCaptor.getValue().getRequestBytes(), StandardCharsets.UTF_8)
+        );
+    }
+
+    @Test
+    public void testsendClusterStateRequest() {
+        ArgumentCaptor<ClusterStateRequest> clusterStateRequestCaptor = ArgumentCaptor.forClass(ClusterStateRequest.class);
+        ClusterStateRequest request = new ClusterStateRequest().clear().indices("foo", "bar");
+        sdkTransportService.sendClusterStateRequest(request);
+        verify(transportService, times(1)).sendRequest(
+            any(),
+            eq(ExtensionsManager.REQUEST_EXTENSION_CLUSTER_STATE),
+            clusterStateRequestCaptor.capture(),
+            any(ClusterStateResponseHandler.class)
+        );
+        assertArrayEquals(new String[] { "foo", "bar" }, clusterStateRequestCaptor.getValue().indices());
+    }
```

Comment:
Do we have a test already for `ClusterStateRequest().all()`?

Reply:
> Do we have a test already for `ClusterStateRequest().all()`?

That's just part of the OpenSearch class.  It sets some booleans to true.  It's the opposite of `clear()` (which sets them to false.  There is a set of ClusterStateRequestTests which randomly set and test the booleans but I don't see one that explicitly tests that they're all set to true (or false). :)